catalog/plugin/catalog-plugin-security-audit/src/main/java/org/codice/alliance/catalog/plugin/security/audit/SecurityAuditPlugin.java

```diff
@@ -0,0 +1,134 @@
+            Metacard updateMetacard = entry.getValue();
+            Metacard existingMetacard = existingMetacards.get(entry.getKey()
+                    .toString());
+            Set<AttributeDescriptor> attributeDescriptors = updateMetacard.getMetacardType()
+                    .getAttributeDescriptors();
+            for (AttributeDescriptor descriptor : attributeDescriptors) {
+                String descriptorName = descriptor.getName();
+                if (auditAttributes.contains(descriptorName) && existingMetacard.getAttribute(
+                        descriptorName) != null) {
+                    Serializable newValue;
+                    if (updateMetacard.getAttribute(descriptorName) != null) {
+                        newValue = updateMetacard.getAttribute(descriptorName)
+                                .getValue();
+                    } else {
+                        newValue = StringUtils.EMPTY_STRING;
+                    }
+                    Serializable oldValue = existingMetacard.getAttribute(descriptorName)
+                            .getValue();
+                    if (!oldValue.equals(newValue)) {
+                        SecurityLogger.audit(String.format(
+                                "%s attribute on metacard %s with value %s was updated to value %s",
+                                descriptorName,
+                                updateMetacard.getId(),
+                                oldValue,
+                                newValue));
+                    }
+                }
+            }
+        }
+
+        return updateRequest;
+    }
+
+    @Override
+    public DeleteRequest processPreDelete(DeleteRequest deleteRequest)
+            throws StopProcessingException {
+        return deleteRequest;
+    }
+
+    @Override
+    public DeleteResponse processPostDelete(DeleteResponse deleteResponse)
+            throws StopProcessingException {
+        return deleteResponse;
+    }
+
+    @Override
+    public QueryRequest processPreQuery(QueryRequest queryRequest) throws StopProcessingException {
+        return queryRequest;
+    }
+
+    @Override
+    public QueryResponse processPostQuery(QueryResponse queryResponse)
+            throws StopProcessingException {
+        return queryResponse;
+    }
+
+    @Override
+    public ResourceRequest processPreResource(ResourceRequest resourceRequest)
+            throws StopProcessingException {
+        return resourceRequest;
+    }
+
+    @Override
+    public ResourceResponse processPostResource(ResourceResponse resourceResponse,
+            Metacard metacard) throws StopProcessingException {
+        return resourceResponse;
+    }
+
+    public void setAuditAttributes(List<String> auditAttributes) {
+        this.auditAttributes = auditAttributes;
+    }
+
+    public static boolean isLocal(Request req) {
+        return req == null || !req.hasProperties() || isLocal((Request) req.getProperties());
```

Comment:
I'm not sure....I think this is a case of a bad copy/paste. I need to look through this again and figure out what happened.